File: 206/ReverseList.go
package _206

import "github.com/raw34/leetcode/runtime"

func reverseList(head *runtime.ListNode) *runtime.ListNode {
	var prev *runtime.ListNode
	curr := head
	for curr != nil {
		next := curr.Next
		curr.Next = prev
		prev = curr
		curr = next
	}

	return prev
}

File: 0206/ReverseList.php
<?php
class ReverseList {
    /**
     * @param ListNode $head
     * @return ListNode
     */
    function reverseList($head) {
        if ($head === null || $head->next === null) {
            return $head;
        }

        $prev = null;

        while ($head) {
            $tmp = $head->next;
            $head->next = $prev;
            $prev = $head;
            $head = $tmp;
        }

        return $prev;
    }
}
